ako-gateway-api/k8s/validator.go

```diff
@@ -245,15 +308,23 @@ func isValidListener(key string, gateway *gatewayv1.Gateway, gatewayStatus *gate
 
 	// Valid listener
 	defaultCondition.
-		Reason(string(gatewayv1.GatewayReasonAccepted)).
+		Reason(string(gatewayv1.ListenerReasonAccepted)).
 		Status(metav1.ConditionTrue).
 		Message("Listener is valid").
 		SetIn(&gatewayStatus.Listeners[index].Conditions)
+
+	// Setting the resolvedRef condition
+	resolvedRefCondition.
+		Status(metav1.ConditionTrue).
+		Reason(string(gatewayv1.ListenerReasonResolvedRefs)).
+		Message("All the references are valid").
+		SetIn(&gatewayStatus.Listeners[index].Conditions)
+
 	utils.AviLog.Infof("key: %s, msg: Listener %s/%s is valid", key, gateway.Name, listener.Name)
 	return true
 }
 
-func IsHTTPRouteValid(key string, obj *gatewayv1.HTTPRoute) bool {
+func IsHTTPRouteValid(key string, obj *gatewayv1.HTTPRoute, gateway *gatewayv1.Gateway) bool {
```

Comment:
need to handle race condition that may occur in this function when both gateway and httproute are trying to validate the same object.

Reply:
have added  locks in validateParentReference and isvalidGateway  methods since they could update status on same gateway.

Reply:
Shouldn't this lock also be added to HTTPRoute validation ?
@arihantg @aaha97

Reply:
have added the lock in HTTPRoute validation as well